datafusion/execution/src/registry.rs

```diff
@@ -34,6 +34,17 @@ pub trait FunctionRegistry {
 
     /// Returns a reference to the udwf named `name`.
     fn udwf(&self, name: &str) -> Result<Arc<WindowUDF>>;
+
+    /// Registers a new `ScalarUDF`, returning any previously registered
```

Comment:
this is a new proposed API -- to allow registering new scalar UDFs with a FunctionRegistry.